source/slang/slang-emit-c-like.cpp

```diff
@@ -2760,32 +2804,44 @@ void CLikeSourceEmitter::_emitInst(IRInst* inst)
 
     case kIROp_SwizzledStore:
         {
-            auto subscriptOuter = getInfo(EmitOp::General);
-            auto subscriptPrec = getInfo(EmitOp::Postfix);
-            bool needCloseSubscript = maybeEmitParens(subscriptOuter, subscriptPrec);
+            switch (this->getTarget())
+            {
+                case CodeGenTarget::CPPSource:
+                case CodeGenTarget::HostCPPSource:
+                case CodeGenTarget::CUDASource:
+                    // cpp and cuda target don't support swizzle on the left handside, so we
+                    // have to assign the element one by one.
+                    _emitSwizzleStorePerElement(inst);
+                    break;
+                default:
```

Comment:
You can `return` for the three cases and keep the remaining logic outside of the `switch` to reduce nesting levels.
You can also just call `isCPUTarget(getTargetReq()) || isCUDATarget(getTargetReq())`.